.github/workflows/lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Lint
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11", "3.12"]
```

Comment:
Do we need separate checks for different python versions?

Reply:
I know we only running 3.12 but using a matrix to check for older versions is nice. If we ever want to upgrade to 3.13 in future we can just add it to the matrix to see if we're compatible. Obviously with linting its a little silly, but the idea is for the long term to add builds and tests to this